build_resolvers/lib/src/resolver.dart

```diff
@@ -284,14 +283,16 @@ Future<void> _createDepsFile(
   await depsFile.writeAsString(jsonEncode(currentDeps));
 }
 
-List<int> _buildSdkSummary(String dartSdkPath) {
+Future<List<int>> _buildSdkSummary(String dartSdkPath) async {
```

Comment:
Lets just just the new private `_runningDartSdkPath` variable since that is all that we pass in here anyways - then we cal also remove the argument from `isFlutterSdk` and the null handling there as well.

Reply:
sg. give me a few.

nit picking, but how do you feel about getting rid of the asynchrony on `isFlutterSdk`, if we're willing to use an `existsSync` in place of the `exists`?

For what it's worth, I've noticed there's a lint that discourages use of `exists`, but that might have baked in assumptions about where the code is running (and whether the files are truly local).

Reply:
Also, happy to keep the version string check. We can just `||` it.

Reply:
I would be fine with using `existsSync` in `isFlutterSdk`. We should also convert that method to a final variable in that case, so we only do it once. Final variables are lazily evaluated so it will only happen on first access.

> Also, happy to keep the version string check. We can just || it.

Ya, lets go ahead and do this as it shouldn't hurt and might be more future proof :).

Reply:
Alright, done, and updated the commit/PR description. PTAL.